cli/command.go

```diff
@@ -0,0 +1,295 @@
+				Name:  noShellFlagName,
+				Usage: "when specified execute commands directly without shell. If multiple commands are specified they do not share a shell environment",
+			},
+			cli.StringSliceFlag{
+				Name:  tagFlagName,
+				Usage: "specify one or more tag names for the process",
+			},
+			cli.BoolFlag{
+				Name:  waitFlagName,
+				Usage: "specify to block until the process returns (subject to service timeouts), propogating the exit code from process",
+			},
+		),
+		Before: mergeBeforeFuncs(clientBefore(),
+			func(c *cli.Context) error {
+				if len(c.StringSlice(commandFlagName)) == 0 {
+					if c.NArg() == 0 {
+						return errors.New("must specify a command")
+					}
+					return errors.Wrap(c.Set(commandFlagName, strings.Join(c.Args().Tail(), " ")), "problem setting command")
+				}
+				return nil
+			}),
+		Action: func(c *cli.Context) error {
+			envvars := c.StringSlice(envFlagName)
+			cmds := c.StringSlice(commandFlagName)
+			useSudo := c.Bool(sudoFlagName)
+			sudoAs := c.String(sudoAsFlagName)
+			useShell := c.BoolT(noShellFlagName)
+			cmdID := c.String(idFlagName)
+			tags := c.StringSlice(tagFlagName)
+			wait := c.Bool(waitFlagName)
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			return withConnection(ctx, c, func(client jasper.RemoteClient) error {
+				cmd := client.CreateCommand(ctx).Sudo(useSudo).ID(cmdID).SetTags(tags)
+
+				for _, cmdStr := range cmds {
+					if useShell {
+						cmd.Bash(cmdStr)
+					} else {
+						cmd.Append(cmdStr)
+					}
+				}
+
+				if sudoAs != "" {
+					cmd.SudoAs(sudoAs)
+				}
+
+				for _, e := range envvars {
+					parts := strings.SplitN(e, "=", 2)
+					cmd.AddEnv(parts[0], parts[1])
+				}
+
+				if err := cmd.Run(ctx); err != nil {
+					return errors.WithStack(err)
+				}
+
+				if wait {
+					exit, err := cmd.Wait(ctx)
+					grip.Notice(err)
+					os.Exit(exit)
+				}
+
+				return nil
+			})
+		},
+	}
+}
+
+// ListCMD provides a user interface to inspect processes managed by a
+// jasper instance and their state. The output of the command is a
+// human-readable table.
+func ListCMD() cli.Command {
+	return cli.Command{
+		Name:  "list",
+		Usage: "list jasper managed commands with human readable output",
+		Flags: append(clientFlags(),
+			cli.BoolFlag{
+				Name:  "running",
+				Usage: "show only running processes",
+			},
+			cli.BoolFlag{
+				Name:  "terminated",
+				Usage: "show only terminated (complete) processes",
+			},
+			cli.BoolFlag{
+				Name:  "failed",
+				Usage: "show only failed processes",
+			},
+			cli.BoolFlag{
+				Name:  "successful",
+				Usage: "show only successful processes",
+			},
+			cli.StringFlag{
+				Name:  "group",
+				Usage: "specify a tag to return a list of processes, overrides other options",
+			},
+		),
+		Before: clientBefore(),
+		Action: func(c *cli.Context) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			filter := jasper.Filter("all")
```

Comment:
This should just be `jasper.All`. I think the `filter.go` needs to expliclty put `Filter` in front of the assignment for the const filters because Go is dumb.